custom-script/utils.py

```diff
@@ -166,6 +223,32 @@ def plot_comparison(rgb_array, sh_prediction, model_prediction, threshold=0.5):
     ax[1][2].set_title("[d] water prediction with model, locally", fontsize=14)
 
     ax[1][3].imshow(sh_thr - model_thr, vmin=-1, vmax=1, cmap="RdBu")
-    ax[1][3].set_title("differences between [d] and [d]", fontsize=14)
+    ax[1][3].set_title("differences between [c] and [d]", fontsize=14)
+
+    plt.tight_layout(pad=0.4, w_pad=0.5, h_pad=2.0)
+
+
+def plot_eopatch(patch: EOPatch, factor: float = 3.15):
+    fig, ax = plt.subplots(ncols=3, figsize=(22, 7))
+    ax[0].imshow(factor * patch.data["BANDS-S2-L1C"][0][..., [3, 2, 1]].squeeze())
+    ax[0].set_title(f"True color, {patch.timestamp[0]}")
+
+    ax[1].imshow(patch.data_timeless["DEM"].squeeze())
+    ax[1].set_title("DEM")
+
+    ax[2].imshow(patch.mask_timeless["water_label"].squeeze(), vmin=0, vmax=1)
+    ax[2].set_title("water mask")
+
+
+def plot_miss_prediction(patch: EOPatch, model_prediction: np.ndarray, threshold: float = 0.5):
+    fig, ax = plt.subplots(ncols=3, figsize=(22, 7))
+    mask = patch.mask_timeless["water_label"].squeeze()
+    ax[0].imshow(mask, vmin=0, vmax=1)
+    ax[0].set_title("water mask")
+
+    model_thr = np.where(model_prediction > threshold, 1, 0)
+    ax[1].imshow(model_thr, vmin=0, vmax=1)
+    ax[1].set_title("model")
 
-    plt.tight_layout(pad=0.05)
+    ax[2].imshow(mask - model_thr, vmin=-1, vmax=1, cmap="RdBu")
+    ax[2].set_title("miss prediction")
```

Comment:
these two are very specific and use hard-coded feature names. Consider moving to notebook or generalize